src/components/product/ProductForm.client.jsx

```diff
@@ -25,6 +28,61 @@ export function ProductForm({productData}) {
     selectedVariant?.priceV2?.amount <
       selectedVariant?.compareAtPriceV2?.amount || false;
 
+  const [socialCount, setSocialCount] = useState();
+  async function setWishlistSocialCount(skipCache){
+    try {
+        const productGQLId = productData.product.id;
+        const productId = productGQLId.split("/")[4];
+        const localCahce = getSwymLocalStorage();
+        const cachedProducts = localCahce.products;
+        const productInfo = cachedProducts && cachedProducts.find(product => product.id == productId);
+
+        if(!skipCache){
+          if(productInfo){
+            setSocialCount(productInfo.count);
+            return;
+          }
+        }
+
+        const res = await getWishlistSocialCount({ empi:productId });
+        if(res?.data?.count){
+          const newCount = res.data.count;
+          setSocialCount(newCount);
+          if(cachedProducts){
+            if(productInfo){
+              const idx = cachedProducts.findIndex(product => product.id == productId);
```

Comment:
@gouravswym you are parsing array twice and you can remove 1 array parsing by 

In line no. 38 you can do `findIndex` and store it then use same here as well.
Will that be ok?